app_dart/analysis_options.yaml

```diff
@@ -42,155 +42,70 @@ analyzer:
-    - prefer_contains
-    # - prefer_double_quotes # opposite of prefer_single_quotes
-    - prefer_equal_for_default_values
-    # - prefer_expression_function_bodies # conflicts with https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#consider-using--for-short-functions-and-methods
-    - prefer_final_fields
-    # - prefer_final_in_for_each # not yet tested
-    - prefer_final_locals
-    # - prefer_for_elements_to_map_fromIterable # not yet tested
-    - prefer_foreach
-    # - prefer_function_declarations_over_variables # not yet tested
-    - prefer_generic_function_type_aliases
-    # - prefer_if_elements_to_conditional_expressions # not yet tested
-    - prefer_if_null_operators
-    - prefer_initializing_formals
-    - prefer_inlined_adds
-    # - prefer_int_literals # not yet tested
-    # - prefer_interpolation_to_compose_strings # not yet tested
-    - prefer_is_empty
-    - prefer_is_not_empty
-    - prefer_iterable_whereType
-    # - prefer_mixin # https://github.com/dart-lang/language/issues/32
-    # - prefer_null_aware_operators # disable until NNBD, see https://github.com/flutter/flutter/pull/32711#issuecomment-492930932
-    - prefer_relative_imports
-    - prefer_single_quotes
-    - prefer_spread_collections
-    - prefer_typing_uninitialized_variables
-    - prefer_void_to_null
-    # - provide_deprecation_message # not yet tested
-    # - public_member_api_docs # enabled on a case-by-case basis; see e.g. packages/analysis_options.yaml
-    - recursive_getters
-    - slash_for_doc_comments
-    # - sort_child_properties_last # not yet tested
-    - sort_constructors_first
-    - sort_pub_dependencies
-    - sort_unnamed_constructors_first
-    - test_types_in_equals
-    - throw_in_finally
-    # - type_annotate_public_apis # subset of always_specify_types
-    - type_init_formals
-    - unawaited_futures
-    # - unnecessary_await_in_return # not yet tested
-    - unnecessary_brace_in_string_interps
-    - unnecessary_const
-    - unnecessary_getters_setters
-    # - unnecessary_lambdas # has false positives: https://github.com/dart-lang/linter/issues/498
-    - unnecessary_new
-    - unnecessary_null_aware_assignments
-    - unnecessary_null_in_if_null_operators
-    - unnecessary_overrides
-    - unnecessary_parenthesis
-    - unnecessary_statements
-    - unnecessary_this
-    - unrelated_type_equality_checks
-    # - unsafe_html # not yet tested
-    - use_full_hex_values_for_flutter_colors
-    # - use_function_type_syntax_for_parameters # not yet tested
-    - use_rethrow_when_possible
-    # - use_setters_to_change_properties # not yet tested
-    # - use_string_buffers # has false positives: https://github.com/dart-lang/sdk/issues/34182
-    # - use_to_and_as_if_applicable # has false positives, so we prefer to catch this by code-review
-    - valid_regexps
-    # - void_checks # not yet tested
+    # https://dart-lang.github.io/linter/lints/index.html
```

Comment:
We should remove all the rules below as we want to move to using the recommended rules.

After removing, I think dart fix will get most of the trivial issues

Reply:
Thanks for the reply! Umm some of the rules were stated in the comments section that they prefer to be excluded, like for avoid_private_typedef_functions, the comment says `we prefer having typedef (discussion in https://github.com/flutter/flutter/pull/16356)`. Should I go ahead and remove these ones as well?

Reply:
Yes, these are a copy and paste from flutter/flutter's repo before we had package:lint to enforce the style guide.